### DISCUSSION TITLE
Indexing excludedPatterns in Neovim

### DISCUSSION THREAD

**Discussion by gacha** [Shopify/ruby-lsp#2700] [Q&A] [answered]
Hi, 
How I understand from [the documentation](https://shopify.github.io/ruby-lsp/editors.html#nvim-lspconfig) then the content of `init_options` is passed to the `ruby-lsp` and I can use any of the [initialization options](https://shopify.github.io/ruby-lsp/editors.html#all-initialization-options). But when I try to set the `indexing.excludedPatterns`, I don't see that this configuration has been picked up.

```lua
      lspconfig.ruby_lsp.setup {
        capabilities = capabilities,
        init_options = {
          formatter = 'standard',
          linters = { 'standard' },
          indexing = {
            excludedPatterns = { '**/tmp/**.rb'  }
          },
        },
      ...
```

What am I missing? How can I debug this?

Thanks in advance!

**Comment by vinistock** [comment, marked as answer]
Hello!

Is that exactly the glob pattern you are using? If it is, there's a small mistake in it. Using `**.rb` is probably not going to match Ruby files, but directories that end with `.rb`. You need the pattern to be

```
**/tmp/**/*.rb
```

**Comment by andyw8** [comment]
@gacha I'm curious, what kind of `.rb` files do you have in `tmp`? I'm wondering if we should be excluding this by default.

**Reply by vinistock** [reply]
We are excluding it https://github.com/Shopify/ruby-lsp/blob/4fd4be9af67c1600e8ad1d762bc4ac87b74ec540/lib/ruby_indexer/lib/ruby_indexer/configuration.rb#L31.

**Comment by gacha** [comment]
Hi, 🤦 sorry. I don't know how I miss that. Now, when I fix the pattern to `'**/tmp/**/*.rb,'` it finally works. Without this, the index finds files in some nested build directory like `tmp/build/some_dir/more_dir/some_file.rb`.

Thanks for your help.

**Reply by gacha** [reply]
I thought I did not come up with that pattern. I found it here https://github.com/Shopify/ruby-lsp/blob/main/vscode/README.md#indexing-configuration and presumed that was the correct syntax.

**Reply by vinistock** [reply]
That was a typo https://github.com/Shopify/ruby-lsp/pull/2708. I'm glad it's working!

**Comment by andyw8** [comment]
> Without this, the index finds files in some nested build directory like tmp/build/some_dir/more_dir/some_file.rb.

Hmm, that's strange. I tried to reproduce by adding a file nested within a directory under `tmp/`, but it wasn't indexed. (in VS Code, didn't try NeoVim).
